backend/src/summary.ts

```diff
@@ -1,50 +1,79 @@
-import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from 'aws-lambda';
-import { Lambda } from "aws-sdk"
-import { ISummarizationData } from "./interface";
+import {
+  APIGatewayProxyEvent,
+  APIGatewayProxyResult,
+  Context,
+} from "aws-lambda";
+import { Lambda, S3 } from "aws-sdk";
+import { CORS_HEADER } from "./constants";
 
 const lambda = new Lambda();
+const s3 = new S3();
 
-export const handler = async (event: APIGatewayProxyEvent, _context: Context): Promise<APIGatewayProxyResult> => {
+export const handler = async (
+  event: APIGatewayProxyEvent,
+  _context: Context
+): Promise<APIGatewayProxyResult> => {
+  let body;
   try {
-    const body = JSON.parse(event.body)
+    body = JSON.parse(event.body);
+  } catch {
+    body = event.body;
+  }
+
+  if (!body || !body.content || !body.length || !body.summaryId) {
+    return {
+      statusCode: 400,
+      headers: CORS_HEADER,
+      body: JSON.stringify(
+        "Invalid Request - Please supply some text and length to summarize."
+      ),
+    };
+  }
 
-    if (!body || !body.content || !body.length) {
-      return {
-        statusCode: 400,
-        body: JSON.stringify("Invalid Request - Please supply some text and length to summarize.")
-      }
-    }
-    
-    const response = await lambda.invoke({
-      FunctionName: 'Essentially-Summary-BE-summaryfunction',
-      InvocationType: 'RequestResponse',
+  const response = await lambda
+    .invoke({
+      FunctionName: "Essentially-Summary-BE-summaryfunction",
+      InvocationType: "RequestResponse",
       Payload: JSON.stringify({
         content: {
           text: body.content,
         },
         length: body.length,
-      })
-    }).promise()
-
-    const payload = JSON.parse(response.Payload.toString())
+      }),
+    })
+    .promise();
 
-    const status = payload.statusCode
+  const payload = JSON.parse(response.Payload.toString());
+  const status = payload.statusCode;
 
-    if (status === 200) {
-        return {
-            statusCode: 200,
-            body: JSON.stringify({ summarized_text: payload.summarized_text, meta: payload.Meta })
-        }
-    } else {
-        return {
-            statusCode: status || 500,
-            body: payload.message 
-        }
-    }
-  } catch {
+  if (status === 200) {
+    await s3
+      .putObject({
+        Bucket:
+          "essentially-backend-s3st-essentiallybackendtestbu-1e6w7ixe3uqi0",
+        Key: body.summaryId.concat("/summary.json"),
+        Body: JSON.stringify({
+          summarized_text: payload.summarized_text,
+          meta: payload.Meta,
+        }),
+        ContentType: "application/json",
+      })
+      .promise();
+    return {
```

Comment:
If we are fetching the summary from s3, is it still necessary to return a body with the summarized text?